packages/caravan-health/src/privacy.test.ts

```diff
@@ -0,0 +1,258 @@
+      ).toFixed(3);
+      expect(score).toBe(0.818);
+    });
+  });
+
+  describe("addressReuseFactor", () => {
+    it("UTXOs having same addresses", async () => {
+      const utxos: AddressUtxos = {
+        address1: [
+          {
+            txid: "tx1",
+            vout: 0,
+            value: 1,
+            status: { confirmed: true, block_time: 0 },
+          },
+          {
+            txid: "tx2",
+            vout: 0,
+            value: 2,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+        address2: [
+          {
+            txid: "tx3",
+            vout: 0,
+            value: 3,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+      };
+      const factor: number = +(
+        await addressReuseFactor(utxos, mockClient)
+      ).toFixed(3);
+      expect(factor).toBe(0);
+    });
+  });
+
+  describe("addressTypeFactor", () => {
+    it("P2PKH address", () => {
+      const transactions = [
+        {
+          vin: [
+            {
+              txid: "input1",
+              vout: 0,
+              witness: [],
+              sequence: 0,
+            },
+          ],
+          vout: [
+            { scriptPubkeyHex: "", scriptPubkeyAddress: "1234", value: 0 },
+          ],
+          txid: "",
+          size: 0,
+          weight: 0,
+          fee: 0,
+          isSend: false,
+          amount: 0,
+          blocktime: 0,
+        },
+      ];
+      const walletAddressType: MultisigAddressType = "P2PKH";
+      const network: Network = Network["MAINNET"];
+      const factor: number = +addressTypeFactor(
+        transactions,
+        walletAddressType,
+        network,
+      ).toFixed(3);
+      expect(factor).toBe(1);
+    });
+  });
+
+  describe("utxoSpreadFactor", () => {
+    it("UTXOs spread across multiple addresses", () => {
+      const utxos = {
+        address1: [
+          {
+            txid: "tx1",
+            vout: 0,
+            value: 1,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+        address2: [
+          {
+            txid: "tx2",
+            vout: 0,
+            value: 2,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+      };
+      const factor: number = +utxoSpreadFactor(utxos).toFixed(3);
+      expect(factor).toBe(0.333);
+    });
+  });
+
+  describe("utxoSetLengthScore", () => {
+    it("UTXO set length", () => {
+      const utxos = {
+        address1: [
+          {
+            txid: "tx1",
+            vout: 0,
+            value: 1,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+        address2: [
+          {
+            txid: "tx2",
+            vout: 0,
+            value: 2,
+            status: { confirmed: true, block_time: 0 },
+          },
+        ],
+      };
+      const score: number = +utxoSetLengthScore(utxos).toFixed(3);
```

Comment:
That is because the signature of `toFixed()` is designed such that it takes in a `number` and outputs a `string`. 

```javascript
    /**
     * Returns a string representing a number in fixed-point notation.
     * @param fractionDigits Number of digits after the decimal point. Must be in the range 0 - 20, inclusive.
     */
    toFixed(fractionDigits?: number): string;
```

Maybe if this looks odd, I can change it with something that rounds floats in number type rather than a string.

Reply:
Ok, this might be fine then. I guess I'm just wondering if this/these functions should just be returning a number full stop so that it won't be necessary to run conversions on it when being interacted with?